build.gradle

```diff
@@ -697,6 +708,21 @@ static def toCamelCase(String input) {
   return input.split("[^a-zA-Z0-9]").collect { it.capitalize() }.join("")
 }
 
+def static getTestClassesForTask(subProject, testTask) {
+  def testDirRoot = "/${testTask.name}/java/"
```

Comment:
Use `sourceSets[${task.name}].srcDirs`? Reason being that you can possibly have tests under `groovy` (or `kotlin`) as well.